env0/utils.go

```diff
@@ -71,13 +71,22 @@ func readResourceData(i interface{}, d *schema.ResourceData) error {
 			continue
 		}
 
+		// custom field is a pointer.
 		if customField, ok := field.Interface().(CustomResourceDataField); ok {
 			if err := customField.ReadResourceData(fieldNameSC, d); err != nil {
 				return err
 			}
 			continue
 		}
 
+		// custom field is a value, check the pointer.
+		if customField, ok := field.Addr().Interface().(CustomResourceDataField); ok {
```

Comment:
what happens when both those checks are not ok? will there be an error? or is it not actually possible?

Reply:
It's possible. In most cases, both will return false. (These cases are uncommon).
It will just continue the function run.